ancestrymanager/ancestryutil.go

```diff
@@ -0,0 +1,96 @@
+package ancestrymanager
+
+import (
+	"fmt"
+	"strings"
+
+	resources "github.com/GoogleCloudPlatform/terraform-validator/converters/google/resources"
+)
+
+// AssetParent derives a resource's parent from its ancestors.
+func AssetParent(cai *resources.Asset, ancestors []string) (string, error) {
+	if err := validateAncestors(ancestors); err != nil {
+		return "", err
+	}
+	if cai == nil {
+		return "", fmt.Errorf("asset not provided")
+	}
+	switch cai.Type {
+	case "cloudresourcemanager.googleapis.com/Folder":
+		if len(ancestors) < 2 {
+			return "", fmt.Errorf("unexpected ancestors %s", ancestors)
```

Comment:
"unexpected ancestors" sounds like the list of ancestors contains some ancestors that were not expected. Suggestion for here and other places:

```suggestion
			return "", fmt.Errorf("unexpected value for ancestors: %s", ancestors)
```

Reply:
Done